From 1990a94554bbe9987a97e08936f1ef9bd5cbcd36 Mon Sep 17 00:00:00 2001
From: Radu Pirea <radu-nicolae.pirea@nxp.com>
Date: Mon, 8 Aug 2022 12:44:56 +0300
Subject: [PATCH 21/42] mk: use gawk instead of mawk

Use gawk instead of mawk.
fip.s32 generated with mawk is broken.

Issue: ALB-9160

Upstream-Status: Pending 

Signed-off-by: Radu Pirea <radu-nicolae.pirea@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/nxp/s32/s32_common.mk | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/plat/nxp/s32/s32_common.mk b/plat/nxp/s32/s32_common.mk
index f2c6d3f77..0287eebef 100644
--- a/plat/nxp/s32/s32_common.mk
+++ b/plat/nxp/s32/s32_common.mk
@@ -16,6 +16,7 @@ ERRATA_A53_1530924	:= 1
 ERRATA_SPECULATIVE_AT	:= 1
 
 # Tools
+AWK ?= gawk
 HEXDUMP ?= xxd
 SED ?= sed
 
@@ -234,7 +235,7 @@ ${FIPTOOL} update --align ${FIP_ALIGN} --tb-fw $1 --soc-fw-config $2 --tb-fw-cer
 endef
 
 define get_fip_hdr_size
-printf "0x%x" $$(${FIPTOOL} info $1 | awk -F'[=,]' '{print strtonum($$2)}' | sort -n | head -n1)
+printf "0x%x" $$(${FIPTOOL} info $1 | ${AWK} -F'[=,]' '{print strtonum($$2)}' | sort -n | head -n1)
 endef
 
 define get_bl2_size
@@ -280,7 +281,7 @@ ${DUMMY_FIP_S32}: ${DUMMY_FIP}
 
 ${IVT_LOCATION_FILE}: ${DUMMY_FIP_S32}
 	${Q}${ECHO} "  MKIMAGE $@"
-	${Q}${MKIMAGE} -l $< 2>&1 | grep 'IVT Location' | awk -F':' '{print $$2}' | xargs > $@
+	${Q}${MKIMAGE} -l $< 2>&1 | grep 'IVT Location' | ${AWK} -F':' '{print $$2}' | xargs > $@
 
 FIP_OFFSET_DELTA ?= 0
 
@@ -291,7 +292,7 @@ endef
 
 ${FIP_OFFSET_FILE}: ${DUMMY_FIP_S32}
 	${Q}${ECHO} "  MKIMAGE $@"
-	${Q}OFF=$$(${MKIMAGE} -l $< 2>&1 | grep Application | awk '{print $$3}');\
+	${Q}OFF=$$(${MKIMAGE} -l $< 2>&1 | grep Application | ${AWK} '{print $$3}');\
 	$(call save_fip_off, $${OFF},$@)
 
 ifndef FIP_HDR_SIZE
-- 
2.17.1

